R/submit-model.R

```diff
@@ -95,17 +96,20 @@ submit_nonmem_model <- function(.mod,
   # define working directory
   model_dir <- get_model_working_directory(.mod)
 
+  coll = makeAssertCollection()
+  assert_file_exists(
+    .config_path %||% file.path(model_dir, "bbi.yaml"),
+    add = coll)
+  if(!coll$isEmpty()){
+    stop(paste("No bbi.yaml file was found in the execution directory.",
+               "Please run `bbi_init()` in the appropriate directory to continue."))
+  }
```

Comment:
The custom error message is a nice improvement.

What's the advantage of this whole `makeAssertCollection` dance over a plain `file_exists` check?   Something like:

```R
if(!fs::file_exists(.config_path %||% file.path(model_dir, "bbi.yaml"))) {
  stop(paste("No bbi.yaml file was found in the execution directory.",
             "Please run `bbi_init()` in the appropriate directory to continue."))
}
```

Two other minor thoughts (feel free to take or leave):

 * If it's a custom config, it doesn't need to be named "bbi.yaml" so perhaps "No bbi.yaml" -> "No bbi configuration"?
 * "run `bbi_init()` _in_ the appropriate directory" -- very minor, but perhaps _with_ is better, since the caller can pass the directory to that function.

Reply:
I suppose there would be no difference in that case. The purpose was to take advantage of the `assert_file_exists` function, but not have to use its default error message (which isn't that informative). There are a number of cases where we use `assert_that` functions, with no custom error message. Though we could replace them all with if-statements, I like the simplicity of using assert_that functions, but being able to customize the error message. Was hoping this and the other PR where I made that change could be a pattern we start to implement.

Reply:
> The purpose was to take advantage of the assert_file_exists function

But what value is the `assert_file_exists` function adding?  In my opinion, it is more complex and less readable, so that pattern should be buying us something if we decide to use it.

Reply:
> Well its minor, but in the case of that function, it also checks accessing rights of those directories.

Hmm, you're talking about the `access` argument?  You're calling that with the default value, so it's not checking the access:

```
> checkmate::assert_file_exists("/etc/sudoers")
> checkmate::assert_file_exists("/etc/sudoers", access = "r")
Error: Assertion on '"/etc/sudoers"' failed: '/etc/sudoers' not readable.
```

> I might be a little biased, but the more i've used checkmate functions, i've really taken a liking to them, partly because I find them less prone to coding errors, but mainly because I like the pattern of them (i.e. always add an assert_blah call for every function argument).

I'm not arguing against checkmate functions for simple argument checking.  I'm objecting to this pattern unless it brings some concrete value that we decide it is worth the added complexity:


```
  coll = makeAssertCollection()
  assert_*(..., add = coll)
  if(!coll$isEmpty()){
    stop(...)
  }
```

To understand that, a reader has to stop and try to figure out `makeAssertCollection` is, what the `add` argument is about, what `$isEmpty()` is referring to.  They'd need to spend much less time parsing this:

```
if(!fs::file_exists(...))) {
  stop(...)
}
```

And if we do decide there is a `checkmate` function that we want to use, why not use the `check_*` variants:

```
> isTRUE(checkmate::check_file_exists("/etc/sudoers"))
[1] TRUE
> isTRUE(checkmate::check_file_exists("/etc/sudoers", access = "r"))
[1] FALSE
```

Reply:
Oop you're right about the access thing. I personally dont think inspecting functions like that is that big of a deal, and comes with the territory, but I realize it looks a little more complicated than it needs to. But again, I know I am biased and like that method, mainly because you can do other things like `coll$push("Custom error message")`, `coll$getMessages()`, etc., and thought that might be a nice pattern to reinforce in the future. 

Hadn't looked into the difference in `test_file_exists` before, so perhaps thats the best path forward for the time being

Reply:
Changed it to:
```r
  .path_exists <- test_file_exists(
    .config_path %||% file.path(model_dir, "bbi.yaml"),
    access = "rw")
  if(!.path_exists){
    stop(paste("No bbi configuration was found in the execution directory.",
               "Please run `bbi_init()` with the appropriate directory to continue."))
  }
```

Reply:
I think checking for access is an overkill and, if the message isn't going to say anything about, it's inviting confusion.  (And is write access really needed?)  I'd just leave it at `file_exists()`, something that is done a good amount in this code base and is widespread throughout code bases in general (nothing specific to R here).

But I'm not going to block on this.  The change to using `test_file_exists` is good enough for me.  Thanks.